cmd/kubelet/app/server.go

```diff
@@ -423,8 +424,12 @@ func Run(ctx context.Context, s *options.KubeletServer, kubeDeps *kubelet.Depend
 	return nil
 }
 
+// checkPermissions confirms that, on linux, the UID of the kubelet process is 0.  For windows, it will return an error automatically.  For other systems, the behaviour is unspecified.
 func checkPermissions() error {
 	if uid := os.Getuid(); uid != 0 {
+		if runtime.GOOS == "windows" {
+			return fmt.Errorf("not checking UID permissions because we're on windows")
```

Comment:
Is this really an error if we are on Windows?

Reply:
Imo it is because of what ravi said, this shouldn't be called for windows .

Reply:
there are bigger issues to be thought out later, but this is just an incremental step in the right direction of making sure the kubelet logs dont totally confuse people debugging failed windows kubelets :)